controllers/group_controller.js

```diff
@@ -0,0 +1,71 @@
+// import http status codes
+const {
+	BAD_REQUEST,
+	NOT_AUTHORIZED,
+	FORBIDDEN,
+	NOT_FOUND,
+	NOT_ACCEPTABLE
+} = require("../utility/statusCodes");
+// import helper functions
+const { sendError, sendSuccess } = require("../utility/helpers");
+const { ObjectId } = require("mongoose").Types.ObjectId;
+
+module.exports.getGroups = async (req, res) => {
+	let { gid } = req.query;
+	let groups;
+	if (gid) {
+		groups = await Group.findById(gid);
+	} else {
+		groups = await Group.find();
+	}
+
+	return sendSuccess(res, groups);
+};
+
+module.exports.myGroups = async (req, res) => {
+	let groups = await Group.aggregate([
+		{
+			$facet: {
+				"Head Groups": [
+					{
+						$match: { heads: { $in: [ObjectId(req.user.id)] } }
+					}
+				],
+				"Member Groups": [
+					{
+						$match: { members: { $in: [ObjectId(req.user.id)] } }
+					}
+				]
+			}
+		}
+	]);
+	return sendSuccess(res, groups);
+};
+
+module.exports.addGroup = async (req, res) => {
+	let { name, heads, members } = req.body;
+	heads = [...new Set(heads)];
+	members = [...new Set(members)];
+	if ((await User.count({ _id: { $in: heads } })) !== heads.length)
```

Comment:
Can we query this count in a single query? @ritiksr25

Reply:
Maybe create a common set of both _id and then query.

Reply:
Yep , that will save an extra query 👍🏽